File: lab1/lab1.Tests/UnitTest1.cs
using System.Diagnostics;
using System.Reflection;
using lab1;

namespace lab1.Tests
{
    [TestClass]
    public class UnitTest1
    {
        [TestMethod]
        public void TestGetMultiply()
        {
            var array = new float[] { 10.5f, 1.3f, 1.7f, 5f, 13.5f, 5.5f };

            var expected = 240.975f;
            var actual = Class1.Multiply(array);

            Assert.AreEqual(expected, actual, 0.00001f);
        }

        [TestMethod]
        public void TestShiftSequenceRight()
        {
            var array = new float[] { 1, 2, 3, 4, 5 };

            var expected = new float[] { 4, 5, 1, 2, 3 };
            var actual = Class1.ShiftArray(array, 2);

            CollectionAssert.AreEqual(expected, actual);
        }

        [TestMethod]
        public void TestGetMaxEven()
        {
            var array = new int[] { 1, 2, 2, 4, 3, 5 };

            var expected = (2,2);
            var actual = Class1.MaxEven(array);

            Assert.AreEqual(expected, actual);
        }
    }
}

File: lab3/UnitTest1/UnitTest1.cpp
#include "pch.h"
#include "CppUnitTest.h"
#include "../lab3/Class1.h"

using namespace Microsoft::VisualStudio::CppUnitTestFramework;

namespace UnitTest1
{
	TEST_CLASS(UnitTest1)
	{
	public:
		Class1 modudeClass;
		TEST_METHOD(TestShiftIntLeft)
		{
			auto actual = modudeClass.shiftInt(12345, 3);
			auto expected = 34512;

			Assert::AreEqual(expected, actual);
		}

		TEST_METHOD(TestShiftIntRight)
		{
			auto actual = modudeClass.shiftInt(12345, 2);
			auto expected = 45123;

			Assert::AreEqual(expected, actual);
		}

		TEST_METHOD(TestShiftIntWithoutShift)
		{
			auto actual = modudeClass.shiftInt(12345, 0);
			auto expected = 12345;

			Assert::AreEqual(expected, actual);
		}

		TEST_METHOD(TestFibNumber)
		{
			auto actual = modudeClass.fibNumber(5);
			auto expected = 5;

			Assert::AreEqual(expected, actual);
		}

		TEST_METHOD(TestFibNumberWithZero)
		{
			auto actual = modudeClass.fibNumber(0);
			auto expected = 0;

			Assert::AreEqual(expected, actual);
		}

		TEST_METHOD(TestDelDecimalInt)
		{
			auto actual = modudeClass.delDecimalInt(123456, 2, 2);
			auto expected = 1256;

			Assert::AreEqual(expected, actual);
		}

		TEST_METHOD(TestDelDecimalIntWithZero)
		{
			auto actual = modudeClass.delDecimalInt(0, 0, 0);
			auto expected = 0;

			Assert::AreEqual(expected, actual);
		}

		TEST_METHOD(TestEvenSumMatrixEl)
		{
			std::vector<std::vector<int>> matrix{ {1,2,3},{4,5,6},{7,8,9} };
			auto actual = modudeClass.evenSumMatrixEl(matrix);
			auto expected = 1;

			Assert::AreEqual(expected, actual);
		}
		TEST_METHOD(TestEvenSumMatrixElWithouEl)
		{
			std::vector<std::vector<int>> matrix{ { },{ } };
			auto actual = modudeClass.evenSumMatrixEl(matrix);
			auto expected = 0;

			Assert::AreEqual(expected, actual);
		}

		TEST_METHOD(TestNumberRankCount)
		{
			auto actual = modudeClass.numberRankCount(123456);
			auto expected = 6;

			Assert::AreEqual(expected, actual);
		}
	};
}
